### ISSUE TITLE
Server not generated on fresh install

### ISSUE THREAD

**Issue by lectrode** [nathanielc/minecraft-multi-server#2]
Thank you for this useful tool. I've been using it without issue for the past year or so.

I fresh-installed Manjaro and am attempting to set this up. However, it seems to be failing to generate the server files. The same issue can be reproduced on hardware or running in a VM.

Steps to reproduce issue:
1) Fresh install latest stable Manjaro (17.1.11 at time of writing)
2) Install `minecraft-multi-server` using makepkg or AUR helper of choice (current version: 0.1.6-1)
3) Restart
4) Create `/etc/minecraft/myworld`:
```
PORT=25599
VERSION=1.12.2
EULA=true
```
5) Run `sudo systemctl start minecraftd@myworld`

Expected result: it will generate `/srv/minecraft/myworld` and populate the folder with necessary files and (attempt to) start the server.

Actual result: it generates the `/srv/minecraft/myworld` folder, but nothing else.

I was able to get the following logs from docker by repeatedly requesting them until they were present:
```
$ sudo docker logs myworld
/start: line 10: eula.txt: Permission denied
/start: line 11: eula.txt: Permission denied
ERROR: unable to write eula to /data. Please make sure attached directory is writable by uid=619
```

As far as I know, permissions are correct. To be certain, I ran the following. Even with this it does not generate the server:
```
sudo chown -R minecraft:minecraft /srv/minecraft
sudo chmod -R 775 /srv/minecraft
```

(Edited for clarity and to remove excess information)

**Comment by lectrode** [comment]
Update: Just to make sure it is a permissions issue, I granted all users full read/write access with
```
sudo chmod -R 777 /srv/minecraft
```

It successfully generates with this. I then changed the permissions back to just the minecraft user having full access, and it fails again:
```
sudo chown -R minecraft:minecraft /srv/minecraft
sudo chmod -R 775 /srv/minecraft
```

I do not know why it does not succeed when only the minecraft user has full access. Obviously, I'd rather not completely open up permissions like this to get it to work properly.

Any ideas?

**Comment by nathanielc** [comment]
Glad to here this is useful to someone besides myself :)

My first thought is that maybe the upstream Docker image changed somehow. That would explain why it continues to work on existing systems ( I am still using it ), because they will be using the original docker image unless an explicit docker pull is made. Then new installs would fail because they would get the newest docker image. The docker images used is https://github.com/itzg/dockerfiles/tree/master/minecraft-server 

I can confirm that the docker image I am using is several months old.

**Comment by lectrode** [comment]
This might be the issue. It looks like the docker was changed to support some other tools, and there's a new setting specifically for the user you intend to run it with:

https://github.com/itzg/dockerfiles/issues/213

I did try adding `-u minecraft` to the docker launch command in minecraftctl, but it didn't seem to make a difference. It is entirely possible I'm missing something, or that might not even be related (although the initial issue report sounds very similar)

**Comment by nathanielc** [comment]
@lectrode FWIW I pulled the new image from  3 days ago and it worked no issue. Maybe they fixed it on  their end? Do you want to try again?

**Comment by lectrode** [comment]
I can confirm that the latest image works. I'm not entirely sure how to _just_ update the image. I ended up deleting everything in `/var/lib/docker`, reinstalling docker and minecraft-multi-server, and rebooting before i could get it working.

However, my process was a bit more complicated, since it turns out i'm also affected by [this bug](https://github.com/moby/moby/issues/27653) since I'm using Btrfs. Docker uses subvolumes on Btrfs, which is cool, except it doesn't clean them up when you remove containers, resulting in more and more space being used up. Not only that, just deleting the `/var/lib/docker` folder can "corrupt" the Btrfs partition (supposedly...I tried this in a VM and even root could not delete the folders (using `rm -rf`) where the subvolumes were mounted until after I deleted the subvolumes, so it didn't let me break my system. it's possible that there is unseen corruption somewhere despite this).

So in addition to the steps above, I also followed some of [this guide](https://gist.github.com/hopeseekr/cd2058e71d01deca5bae9f4e5a555440) for cleaning up the mess docker made on Btrfs and using a mounted ext4 image specifically for /var/lib/docker.

Closing this issue since the original problem has been taken care of upstream. Thank you for your assistance!
